tests/smoke/test_run_s2p.py

```diff
@@ -0,0 +1,33 @@
+import suite2p
+
+
+def test_do_registration_do_roi_detect_settings_check_timing(test_ops):
+    """
+    Tests that do_registration and roidetect parameters don't error. Also, makes sure timing_report
+    doesn't error and checks timing dictionary to make sure the times for only the stages specified are provided.
+    """
+    test_ops.update({
+        'tiff_list': ['input.tif'],
+        'roidetect': False,
+        'do_registration': False,
+        'spikedetect': False,
+    })
+    conv_only_ops = suite2p.run_s2p(ops=test_ops)  # conversion only
+    assert list(conv_only_ops[0]['timing'].keys()) == ['total_plane_runtime']
+    test_ops.update({
+        'do_registration': True
+    })
+    reg_only_ops = suite2p.run_s2p(ops=test_ops)  # registration only
+    assert list(reg_only_ops[0]['timing'].keys()) == ['registration', 'total_plane_runtime']
+    test_ops.update({
+        'do_registration': False,
+        'roidetect': True,
+    })
+    det_only_ops = suite2p.run_s2p(ops=test_ops)  # detection step only
+    assert list(det_only_ops[0]['timing'].keys()) == ['detection', 'extraction', 'classification', 'total_plane_runtime']
+    test_ops.update({
+        'spikedetect': True
+    })
+    det_dec_ops = suite2p.run_s2p(ops=test_ops)  # detection & deconvolution
+    assert list(det_dec_ops[0]['timing'].keys()) == ['detection', 'extraction', 'classification',
+                                                     'deconvolution', 'total_plane_runtime']
```

Comment:
This test is quite complex, and it runs run_s2p() multiple times (generally costly) and only checks that the right key combinations appear in the ops['timing'] dict.  Maybe this isn't a necessary test? Could be simply removed without much cost.

Reply:
That's not quite the only thing it checks. The primary reason I added this test is to check that the ops params do_registration, roidetect, and spikedetect work as intended. Those are settings that can be modified by the GUI and we didn't have any tests that make sure the pipeline doesn't error when each of those are turned off. This test checks that each stage can proceed properly(e.g: if registration only happens the first run and then the next run starts off with no registration, can the pipeline find the registered binary/necessary ops params and start detection?) and alerts us of any errors that occur in between the stages The timing checks come as an added benefit.

Reply:
As for the multiple runs, registration(which takes up much of the time) only happens once and so this test takes a bit longer than a single 1plane1chan full pipeline test. I did forget to specify a tiff list so it's prolly picking up the input and input1500 tifs. I'll make that change. If it's still too long, we could even use a dataset with even smaller images/fewer frames

Reply:
I think this sort of test is very useful because it is something that can and has broken in the past - and this is a use case that several users will use. It might be even more useful for GUI checking if the CLI is used but as long as we don't change `__main__.py` this test is sufficient